vmdb/app/models/miq_request.rb

```diff
@@ -502,4 +472,11 @@ def self.update_request(request, values, requester_id, target_class, event_messa
     return request
   end
 
+  private
+
+  def request_types
+    request_types = MODEL_REQUEST_TYPES.values.each_with_object({}) { |i, h| i.each { |k, v| h[k] = v } }
+    request_types[:MiqProvisionRequestTemplate] = {:template => "VM Provision Template"}
```

Comment:
Can this just be left as a constant?  As @kbrock said, there's only need to build it once.  You could do it right up there where the constant is defined.